pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -1586,19 +1594,34 @@ public long getBacklogSize() {
         return ledger.getEstimatedBacklogSize();
     }
 
-    public boolean isActive() {
+    public boolean isActive(InactiveTopicDeleteMode deleteMode) {
+        switch (deleteMode) {
+            case delete_when_no_subscriptions:
+                if (!subscriptions.isEmpty()) {
+                    return true;
+                }
+            case delete_when_subscriptions_caught_up:
+                if (hasLocalConsumers() || hasBacklogs()) {
```

Comment:
I think in `delete_when_subscriptions_caught_up` mode, we just need to check if the backlog is empty or not. we don't need to check if there are consumers connected.